tests/units/result_manager/test__init__.py

```diff
@@ -149,6 +171,95 @@ def test_add(
                 assert result_manager.status == expected_status
             assert len(result_manager) == 1
 
+    def test_add_type_error(self) -> None:
+        """Test ResultManager.add with wrong object type."""
+        result_manager = ResultManager()
+        with pytest.raises(TypeError, match="Added test result 'test' must be a TestResult instance, got str."):
+            result_manager.add("test")  # type: ignore[arg-type]
+
+    def test_get_results(self, test_result_factory: Callable[[], TestResult]) -> None:
```

Comment:
maybe this needs to be splitted up - the test is very long

Reply:
Fixed in 73208a5aa2055f69a02daf2e7e18464a30a3983e